### DISCUSSION TITLE
Generators that depend on multiple files

### DISCUSSION THREAD

**Discussion by WarrenWeckesser** [mesonbuild/meson-python#506] [Q&A] [answered]
I'm using generators to process Jinja templates into source code, e.g. `foo.c.jinja` is transformed to `foo.c`.  This works great, but as my Jinja files get more complicated, I find it useful to share common Jinja statements in a file that is included by other Jinja files.  That is, `foo.c.jinja` has the line `{% include "common.jinja" %}`, where `common.jinja` contains Jinja definitions.  I need a way to tell meson that `foo.c` also depends on `common.jinja`, but I don't see how to do this with a generator.  Is this possible, or should I switch to using custom targets?  C.f. https://github.com/mesonbuild/meson/issues/6064

**Comment by rgommers** [comment, marked as answer]
> I need a way to tell meson that `foo.c` also depends on `common.jinja`, but I don't see how to do this with a generator. Is this possible

Hey Warren, I think it may be possible with the `depends` keyword of [generator](https://mesonbuild.com/Reference-manual_functions.html#generator). See for example `linalg_init_cython_gen` in `scipy/linalg/meson.build`. All Cython generators in SciPy depend also on `__init__.py` and `.pxd` files.

**Reply by WarrenWeckesser** [reply]
Thanks Ralf.  I had seen the `depends` keyword, but my naive attempt to use it with just the filename resulted in an error--a simple filename is not a build target.  It works if I use the "copyfile" trick as done in the SciPy code that you pointed out, i.e.:

    fs = import('fs')

    common_jinja = fs.copyfile('common.jinja')

    jinja = generator(<...>,
                      depends : [common_jinja])
